zerver/test_hooks.py

```diff
@@ -953,3 +953,72 @@ def test_yo_message(self):
         self.assertEqual(msg.content, (u"Yo from IAGO"))
+
+
+class CodeshipHookTests(AuthedTestCase):
+    STREAM_NAME = 'codeship'
+    TEST_USER_EMAIL = 'hamlet@zulip.com'
+    URL_TEMPLATE = "/api/v1/external/codeship?stream={stream}&api_key={api_key}"
+
+    def setUp(self):
+        api_key = self.get_api_key(self.TEST_USER_EMAIL)
+        self._url = self.URL_TEMPLATE.format(stream=self.STREAM_NAME, api_key=api_key)
+        self.subscribe_to_stream(self.TEST_USER_EMAIL, self.STREAM_NAME)
+
+    def test_codeship_build_in_testing_status_message(self):
+        """
+        Tests if codeship testing status is mapped correctly
+        """
+        body = self._get_fixture_data('testing_build')
+        self._send_post_request_with_params(body)
+
+        expected_message = u"[Build](https://www.codeship.com/projects/10213/builds/973711) triggered by beanieboi on master branch started."
+        msg = self._get_recently_added_message()
+        self.assertEqual(msg.subject, u"codeship/docs build status.")
+        self.assertEqual(msg.content, expected_message)
+
+    def test_codeship_build_in_error_status_message(self):
+        """
+        Tests if codeship error status is mapped correctly
+        """
+        body = self._get_fixture_data('error_build')
+        self._send_post_request_with_params(body)
+
+        expected_message = u"[Build](https://www.codeship.com/projects/10213/builds/973711) triggered by beanieboi on master branch failed."
+        msg = self._get_recently_added_message()
+        self.assertEqual(msg.subject, u"codeship/docs build status.")
+        self.assertEqual(msg.content, expected_message)
+
+    def test_codeship_build_in_success_status_message(self):
+        """
+        Tests if codeship success status is mapped correctly
+        """
+        body = self._get_fixture_data('success_build')
+        self._send_post_request_with_params(body)
+
+        expected_message = u"[Build](https://www.codeship.com/projects/10213/builds/973711) triggered by beanieboi on master branch succeeded."
+        msg = self._get_recently_added_message()
+        self.assertEqual(msg.subject, u"codeship/docs build status.")
+        self.assertEqual(msg.content, expected_message)
+
+    def test_codeship_build_in_other_status_status_message(self):
+        """
+        Tests if codeship other status is mapped correctly
+        """
+        body = self._get_fixture_data('other_status_build')
+        self._send_post_request_with_params(body)
+
+        expected_message = u"[Build](https://www.codeship.com/projects/10213/builds/973711) triggered by beanieboi on master branch has some_other_status status."
+        msg = self._get_recently_added_message()
+        self.assertEqual(msg.subject, u"codeship/docs build status.")
+        self.assertEqual(msg.content, expected_message)
+
+    def _get_recently_added_message(self):
+        return Message.objects.filter().order_by('-id')[0]
```

Comment:
Seems like it could be good to move this function into a generic helper that lives in test_helpers.py, since it seems like it'd be useful for a lot of tests.  

Changing this should be its own commit but if you grep for `Message.objects.get` and `Message.objects.filter` there's like 10 instances of this in the codebase that we could convert.